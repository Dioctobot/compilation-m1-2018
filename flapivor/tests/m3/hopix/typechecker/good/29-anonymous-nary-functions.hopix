val apply : (Int -> Int) * Int -> Int =
    (fun (f, x) => f (x) : (Int -> Int) * Int -> Int)

def succ : Int -> Int (x) = x + 1

val apply2 : (Int * Int -> Int) * Int * Int -> Int =
    (fun (f, x, y) => f (x, y) : (Int * Int -> Int) * Int * Int -> Int)

def addmul : Int * Int -> Int (x, y) = x * y + y * x

val apply3 : (Int * Int * Int -> Int) * Int * Int * Int -> Int =
    (fun (f, x, y, z) => f (x, y, z) : (Int * Int * Int -> Int) * Int * Int * Int -> Int)

def conj : Int * Int * Int -> Int (x, y, z) = x * y * z

val main : Int =
    apply (succ, -1) +
    apply2 (addmul, 3, 7) +
    apply3 (conj, 0, 1, 2)



