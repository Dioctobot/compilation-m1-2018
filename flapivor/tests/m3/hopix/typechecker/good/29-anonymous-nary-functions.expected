apply : (Int -> Int) * Int -> Int
succ : Int -> Int
apply2 : (Int * Int -> Int) * Int * Int -> Int
addmul : Int * Int -> Int
apply3 : (Int * Int * Int -> Int) * Int * Int * Int -> Int
conj : Int * Int * Int -> Int
main : Int
